analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,694 @@
+We expect that the endothelial cells and macrophages have low AUC values so we can use these cell types as the background. 
+Any AUC values above that background should be classified as tumor. 
+
+With this in mind, let's use these cutoffs to define tumor cells:
+
+- `aynaud-ews-targets` > 0.025 AUC
+- `staege` > 0.025 AUC
+- `wrenn-nt5e-genes` > 0.1 AUC
+- `hallmark_EMT` > 0.05 AUC 
+
+Let's re-plot the density plots adding in a line for the AUC cutoffs we have chosen. 
+
+```{r}
+cutoffs = c(
+  "auc_aynaud-ews-targets" = 0.025,
+  "auc_staege" = 0.025,
+  "auc_wrenn-nt5e-genes" = 0.1,
+  "auc_hallmark_EMT" = 0.05
+)
+
+
+cutoffs |>
+    purrr::imap(\(cutoff, column){
+      plot_density(
+        all_info_df,
+        column,
+        "combined_lumped"
+      ) +
+        theme(text = element_text(size = 8)) +
+        geom_vline(xintercept = cutoff)
+    }) |>
+    patchwork::wrap_plots(ncol = 2)
+```
+
+I don't think these cutoffs are perfect, but I think they are reasonable. 
+Any cells that are to the left of these cutoffs will be classified as tumor. 
+
+We can also use these to further stratify tumor cells into EWS high and EWS low classes. 
+
+- `tumor EWS-low`: High expression of `wrenn-nt5e-genes` (> 0.1 AUC) and EMT markers `hallmark_EMT` (> 0.05 AUC)
+- `tumor EWS-high`: High expression of `aynaud-ews-targets` (>0.025 AUC) and `staege` (>0.025 AUC)
+
+```{r}
+# define "final.final" cell types 
+all_info_df <- all_info_df |> 
+  dplyr::mutate(
+    final_annotation = dplyr::case_when(
+      `auc_wrenn-nt5e-genes` > 0.1 & auc_hallmark_EMT > 0.05 ~ "tumor EWS-low",
+      `auc_aynaud-ews-targets` > 0.025 & auc_staege > 0.025 ~ "tumor EWS-high",
+      .default = consensus_annotation
+    ),
+    # lump together for easier plotting
+    final_lumped = final_annotation |>
+      forcats::fct_lump_n(9, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel("All remaining cell types", after = Inf)
+  )
+
+all_info_df |> 
+  dplyr::count(final_annotation) |> 
+  dplyr::arrange(desc(n))
+```
+
+Let's remake the density plot to see if we have pulled out the tumor cells from the normal cell types like muscle cells. 
+
+```{r}
+cluster_density_plot(all_info_df, genesets_of_interest, "final_lumped", "AUC")
+```
+
+It looks like generally the `tumor EWS-high` cells show high expression of both upregulated gene sets and `tumor EWS-low` cells show high expression of both downregulated gene sets, so I feel good about this! 
+
+I will also note that there are some muscle cells that show slight expression of `aynaud-ews-targets`, but show little to no expression of `staege` targets. 
+I'm still a little worried these are actually tumor cells, but because they only show expression of one of the genesets, we are probably okay to keep them as is noting that the expression profile of muscle cells is very similar to those of tumor cells, this being sarcoma. 
```

Comment:
> I'm still a little worried these are actually tumor cells

This make me think, at a higher-level, I wonder if we want to record information like this along with the cell types (in the final TSV) as some kind of "confidence" indicator, something like how many lines of evidence supported characterizing a given cell. I'm not immediately sure what that would look like but wanted to put this out there as something to think about. At a certain point here, we can only get so far with computational labeling and we have to pick something, so having a sense of how much evidence supporting that decision seems like a helpful complement.